### DISCUSSION TITLE
Running "training" module on CPU

### DISCUSSION THREAD

**Discussion by Mrcel97** [IntelLabs/RAG-FiT#8] [Q&A]
Hello, I'm trying to replicate the full flow of modules but using only CPU as my device. For now I've been able to reach 3 out of the 4 modules being able to "processing", "inference" (without LoRa), and "evaluation".

My question (with ignorance on what the LoRa procedure involves) is regarding this missing "training" module which I cannot run even though applying the patch* I've applied to the 3rd "inference" in order to run on CPU. Do "training"-stage support running on CPU? Or is just a dead-end at this point and not worth spending more time into it?

Thanks in advance!

- Patch: `export PYTORCH_ENABLE_MPS_FALLBACK=1`

**Comment by danielfleischer** [comment]
Sorry I missed this. 

I'm not aware of any conceptual limitation in running training on CPU. What exactly is the error you're getting?
